Lib/ipaddress.py

```diff
@@ -1160,19 +1163,24 @@ def _make_netmask(cls, arg):
             if isinstance(arg, int):
                 prefixlen = arg
                 if not (0 <= prefixlen <= cls._max_prefixlen):
-                    cls._report_invalid_netmask(prefixlen)
+                    cls._report_invalid_netmask(prefixlen, suppress=False)
             else:
-                try:
-                    # Check for a netmask in prefix length form
-                    prefixlen = cls._prefix_from_prefix_string(arg)
-                except NetmaskValueError:
-                    # Check for a netmask or hostmask in dotted-quad form.
-                    # This may raise NetmaskValueError.
-                    prefixlen = cls._prefix_from_ip_string(arg)
+                prefixlen = cls._prefix_from_string(arg)
```

Comment:
It was moved to a separate function because it is hard to write it correctly without `return`. Otherwise we would need to write something like (not tested):

```python
        try:
            prefixlen = cls._prefix_from_prefix_string(arg)
        except NetmaskValueError:
            try:
                prefixlen = cls._prefix_from_ip_string(arg)
            except BaseException as exc:
                raise exc from None
```